src/main/java/net/minecraftforge/fml/common/network/FMLOutboundHandler.java

```diff
@@ -231,6 +233,39 @@ public void validateArgs(Object args)
                 return builder.build();
             }
         },
+        /**
+         * The packet is sent to all players that are watching the Chunk containing the supplied {@link TargetPoint}.
+         * The {@code range} field of the {@link TargetPoint} is ignored.
+         */
+        ALLAROUNDTRACKING(Sets.immutableEnumSet(Side.SERVER))
+        {
+            @Override
+            public void validateArgs(Object args)
+            {
+                if (!(args instanceof TargetPoint))
+                {
+                    throw new RuntimeException("ALLAROUNDTRACKING expects a TargetPoint argument");
+                }
+            }
+
+            @Nullable
+            @Override
+            public List<NetworkDispatcher> selectNetworks(Object args, ChannelHandlerContext context, FMLProxyPacket packet)
+            {
+                TargetPoint tp = (TargetPoint)args;
+                ImmutableList.Builder<NetworkDispatcher> builder = ImmutableList.builder();
+                PlayerChunkMap chunkMap = FMLCommonHandler.instance().getMinecraftServerInstance().getWorld(tp.dimension).getPlayerChunkMap();
+                for (EntityPlayerMP player : FMLCommonHandler.instance().getMinecraftServerInstance().getPlayerList().getPlayers())
+                {
+                    if (player.dimension == tp.dimension && chunkMap.isPlayerWatchingChunk(player, MathHelper.floor(tp.x) >> 4, MathHelper.floor(tp.z) >> 4))
+                    {
+                        NetworkDispatcher dispatcher = player.connection.netManager.channel().attr(NetworkDispatcher.FML_DISPATCHER).get();
+                        if (dispatcher != null) builder.add(dispatcher);
+                    }
+                }
+                return builder.build();
```

Comment:
Instead of iterating all players and checking if they are watching, why not use `PlayerChunkMapEntry#players`?

Reply:
can't access the player list without reflecting or AT-ing it, which I don't know how to do :P

Reply:
Add a patch for a `getPlayers()` method?
```java
    public List<EntityPlayerMP> getPlayers()
    {
        return this.sentToPlayers ? java.util.Collections.unmodifiableList(players) : java.util.Collections.emptyList();
    }
```

Reply:
I don't imagine any performance losses from doing it the current way so I'd rather not add patches

Reply:
Then maybe at least inline `PlayerChunkMap#isPlayerWatchingChunk` and move `PlayerChunkMap#getEntry` + ifs outside loop? No reason to make same map lookup for every player and you can skip whole loop, if entry is not valid.